volatility3/framework/plugins/windows/psxview.py

```diff
@@ -0,0 +1,279 @@
+
+    def translate_offset(self, offset):
+        if self.config["physical-offsets"]:
+            return offset
+
+        kernel = self.context.modules[self.config["kernel"]]
+        layer_name = kernel.layer_name
+
+        try:
+            offset = list(
+                self.context.layers[layer_name].mapping(offset=offset, length=0)
+            )[0][2]
+        except:
+            # already have physical address
+            pass
+
+        return offset
+
+    def proc_list_to_dict(self, tasks):
+        return {self.translate_offset(proc.vol.offset): proc for proc in tasks}
+
+    def check_pslist(self, tasks):
+        res = self.filter_garbage_procs(tasks)
+        return self.proc_list_to_dict(tasks)
+
+    def check_psscan(self, layer_name, symbol_table):
+        res = psscan.PsScan.scan_processes(
+            context=self.context, layer_name=layer_name, symbol_table=symbol_table
+        )
+        res = self.filter_garbage_procs(res)
+
+        return self.proc_list_to_dict(res)
+
+    def check_thrdscan(self):
+        ret = []
+
+        for ethread in thrdscan.ThrdScan.scan_threads(
+            self.context, module_name="kernel"
+        ):
+            process = None
+            try:
+                process = ethread.owning_process()
+                if not process.is_valid():
+                    continue
+
+                ret.append(process)
+            except AttributeError:
+                vollog.log(
+                    constants.LOGLEVEL_VVV,
+                    "Unable to find the owning process of ethread",
+                )
+
+        return self.proc_list_to_dict(ret)
+
+    def check_csrss_handles(self, tasks, layer_name, symbol_table):
+        ret = []
+
+        for p in tasks:
+            name = self.proc_name_to_string(p)
+            if name == "csrss.exe":
+                try:
+                    if p.has_member("ObjectTable"):
+                        handles_plugin = handles.Handles(
+                            context=self.context, config_path=self.config_path
+                        )
+                        hndls = list(handles_plugin.handles(p.ObjectTable))
+                        for h in hndls:
+                            if (
+                                h.get_object_type(
+                                    handles_plugin.get_type_map(
+                                        self.context, layer_name, symbol_table
+                                    )
+                                )
+                                == "Process"
+                            ):
+                                ret.append(h.Body.cast("_EPROCESS"))
+
+                except exceptions.InvalidAddressException:
+                    vollog.log(
+                        constants.LOGLEVEL_VVV, "Cannot access eprocess object table"
+                    )
+
+        ret = self.filter_garbage_procs(ret)
+        return self.proc_list_to_dict(ret)
+
+    def check_session(self, pslist_procs):
```

Comment:
None of these methods are really worth exposing to the outside world, or can be reused easily, so it would be better to prefix them with `_` to show that they're supposed to be private.